### PR TITLE
Update Lambda example

### LINKED ISSUE #537: Confused about documentation
Hello,
I was preparing a lambda function in rust using this runtime but I found a difference between the readme in this project and the documentation by aws at (https://docs.aws.amazon.com/sdk-for-rust/latest/dg/lambda.html).
The readme suggests the use of the service_fn function while the aws docs make use of the handler_fn function.

Can you please explain which is the best way?
Thanks,
Frank

### PR DESCRIPTION
**Pull request by calavera**
This PR updates the Lambda example with a more up to date implementation. It also changes the building section to use Cargo Lambda because it's currently the recommended approach, simplifying that step for customers.

Fixes https://github.com/awslabs/aws-lambda-rust-runtime/issues/537

### CODE REVIEW

**Review by calavera — COMMENTED** [review]
This change aligns with the current documentation in the Lambda Runtime's README and examples: https://github.com/awslabs/aws-lambda-rust-runtime

File: doc_source/lambda.md [hunk 1]
```diff
@@ -39,189 +33,166 @@ license = "YOUR-LICENSE"
...
-   cargo install cross
-   ```
+    // Initialize the client here to be able to reuse it across
+    // different invocations.
+    //
+    // No extra configuration is needed as long as your Lambda has
+    // the necessary permissions attached to its role.
+    let config = aws_config::load_from_env().await;
+    let s3_client = aws_sdk_s3::Client::new(&config);
+    let client_ref = &s3_client;
```
Comment (DavidSouther) [inline doc_source/lambda.md]: Just take the reference at the call site, no need to separate it out.

File: doc_source/lambda.md [hunk 2]
```diff
@@ -39,189 +33,166 @@ license = "YOUR-LICENSE"
...
         .content_type("text/plain")
         .send()
-        .await
-        .map_err(|err| {
+        .await;
+
+    match response {
+        Ok(_) => {
+            tracing::info!(
+                "Successfully stored the incoming request in S3 with the name '{}'",
```
Comment (DavidSouther) [inline doc_source/lambda.md]: `'{filename}'`

File: doc_source/lambda.md [hunk 3]
```diff
@@ -39,189 +33,166 @@ license = "YOUR-LICENSE"
...
+                &filename
+            );
+
+            // Return `Response` (it will be serialized to JSON automatically by the runtime)
+            Ok(Response {
+                req_id: event.context.request_id,
+                body: format!(
+                    "the lambda has successfully stored the your request in S3 with name '{}'",
```
Comment (DavidSouther) [inline doc_source/lambda.md]: Same, '{filename}'

File: doc_source/lambda.md [hunk 4]
```diff
@@ -39,189 +33,166 @@ license = "YOUR-LICENSE"
...
+                ),
+            })
+        }
+        Err(err) => {
             // In case of failure, log a detailed error to CloudWatch.
-            error!(
+            tracing::error!(
                 "failed to upload file '{}' to S3 with error: {}",
-                &filename, err
+                &filename,
```
Comment (DavidSouther) [inline doc_source/lambda.md]: '{filename}', '{err}'

File: doc_source/lambda.md [hunk 5]
```diff
@@ -39,189 +33,166 @@ license = "YOUR-LICENSE"
...
-    let s3_client = aws_sdk_s3::Client::new(&config);
+async fn put_object(
+    s3_client: &aws_sdk_s3::Client,
+    bucket_name: &str,
+    event: LambdaEvent<Request>,
+) -> Result<Response, Error> {
+    tracing::info!("handling a request...");
     // Generate a filename based on when the request was received.
-    let filename = format!("{}.txt", time::OffsetDateTime::now_utc().unix_timestamp());
+    let filename = format!("{}.txt", chrono::offset::Utc::now().timestamp());
```
Comment (DavidSouther) [inline doc_source/lambda.md]: I don't think it's worth pulling in the entire chrono crate for generating a file name. Was std::time not giving the right values for the name?

File: doc_source/lambda.md [hunk 6]
```diff
@@ -39,189 +33,166 @@ license = "YOUR-LICENSE"
...
-}
-
-// Implement Display for the Failure response so that we can then implement Error.
-impl std::fmt::Display for FailureResponse {
+impl std::fmt::Display for Response {
+    /// Display the response struct as a JSON string
     fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
-        write!(f, "{}", self.body)
+        let err_as_json = serde_json::json!(self).to_string();
+        write!(f, "{}", err_as_json)
```
Comment (DavidSouther) [inline doc_source/lambda.md]: mcv is enough for string interpolation, `"{err_as_json}"`

File: doc_source/lambda.md [hunk 7]
```diff
@@ -39,189 +33,166 @@ license = "YOUR-LICENSE"
...
-   ```
-
-If this approach fails, try the next one\.
-
-### Container approach<a name="lambda-step3-x2"></a>
-
-This approach requires that you install either [Docker](https://www.docker.com) or [Podman](https://podman.io)\.
+#[tokio::main]
+async fn main() -> Result<(), Error> {
+    tracing_subscriber::fmt()
```
Comment (DavidSouther) [inline doc_source/lambda.md:144]: I almost want a default tracing_subscriber with these settings for Lambda... I'll copy this to the docs examples, thanks!

(No action needed)

**Review by DavidSouther — CHANGES_REQUESTED** [review]
Thank you, calavera, for taking the time to update this! Looks really solid overall, just a couple nits around using captured identifiers.

**Review by DavidSouther — APPROVED** [review]

**Pull request merged**